packages/connectivity/connectivity_macos/macos/Classes/IPHelper.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+#import "SystemConfiguration/CaptiveNetwork.h"
+
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+
+NSString* getWifiIP() {
+  NSString* address = @"error";
+  struct ifaddrs* interfaces = NULL;
+  struct ifaddrs* temp_addr = NULL;
+  int success = 0;
+
+  // Retrieve the current interfaces - returns 0 on success.
+  success = getifaddrs(&interfaces);
+  if (success == 0) {
+    // Loop through linked list of interfaces.
+    temp_addr = interfaces;
+    while (temp_addr != NULL) {
+      if (temp_addr->ifa_addr->sa_family == AF_INET) {
+        // Check if interface is en0 which is the wifi connection on the iPhone.
+        if ([[NSString stringWithUTF8String:temp_addr->ifa_name] isEqualToString:@"en0"]) {
+          // Get NSString from C String
+          address = [NSString
+              stringWithUTF8String:inet_ntoa(((struct sockaddr_in*)temp_addr->ifa_addr)->sin_addr)];
```

Comment:
Are all of these guaranteed to be non-null? Same question about the if block above.

Reply:
No, but in this case it's not an issue. 

In the block above, calling `isEqualToString` on nil is a no-op, so the if statement would be false.
Passing nil as the return value is also acceptable.